### ISSUE TITLE
Native crash on Nexus 9 with Android 7.1.1

### ISSUE THREAD

**Issue by MaksimDmitriev** [box/box-android-sdk#283]
Hello,

Here is what I did:

1. I installed a debug version of box-content-sample
2. I pressed Log in on the options menu
3. Then I pressed Terms of Service
4. A dialog with the text "Security warning. There are problems with the security certificate..."
5. I pressed Back
6. Then I pressed Log in on the options menu (or closed the app, re-opened it, and then pressed Log in)
7. The app crashed

The app doesn't crash every time like that. What I am sure about is that one needs to press "Terms of Service" and then "Log in" a few times to reproduce the crash. The log file is attached.

I saw /system/app/Chrome/Chrome.apk in my log file. If it matters, the version of Chrome is 53.0.2785.135

Our test engineer says she managed to reproduce the same crash on Samsung GT-I9192 with 4.4.2 and Samsung SM-G900F with Android 6.0.1

[Box Sample crash.txt](https://github.com/box/box-android-sdk/files/1055284/Box.Sample.crash.txt)

**Comment by doncung** [comment]
Thanks Maksim this is helpful. As a first measure we'll see if we can fix the cert issue with our ToS so hopefully after that the crash will be non reproducible (unless there is a real man in the middle scenario).

**Comment by MaksimDmitriev** [comment]
I'm glad I was helpful. 
When do you think your team will fix it?

**Comment by doncung** [comment]
A ticket has already been filed to fix the ToS with the cert issue. Since
that is a different team, it is hard for me to estimate and mostly depends
on what they have on their backlog.

**Comment by doncung** [comment]
One of our server teams is looking into this issue to solve the cert issues. The crash may be due to showing too many alert dialogs as the current logic shows one for each security warning and the ToS site has a number of certs that aren't signed by a natively recognized root authority. I'll look into what we can do on the sdk end, but fixing the cert issue should remove the bug.

**Comment by MaksimDmitriev** [comment]
@doncung  yes, seeing too many dialogs is another issue. Sorry, forgot to report it.
